worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java

```diff
@@ -79,44 +83,51 @@ public Job putJobInProgress(String jobId) {
 
     @Override
     @Transactional(propagation = Propagation.NEVER)
-    public Job processJob(final String jobId) {
+    public Job processJob(final String jobUuid) {
 
-        final Job job = jobRepository.findByJobUuid(jobId);
-        log.info("Found job : {}", job.getJobUuid());
+        final Job job = jobRepository.findByJobUuid(jobUuid);
+        log.info("Found job : {}", jobUuid);
 
         final Sponsor sponsor = job.getUser().getSponsor();
 
         final List<Contract> attestedContracts = sponsor.getAggregatedAttestedContracts();
-        log.info("Job [{}] has [{}] attested contracts", job.getJobUuid(), attestedContracts.size());
+        log.info("Job [{}] has [{}] attested contracts", jobUuid, attestedContracts.size());
 
         try {
-            final Path outputDirPath = Paths.get(efsMount, job.getJobUuid());
+            final Path outputDirPath = Paths.get(efsMount, jobUuid);
             final Path outputDir = fileService.createDirectory(outputDirPath);
 
             for (Contract contract : attestedContracts) {
-                log.info("Job [{}] - contract [{}] ", job.getJobUuid(), contract.getContractNumber());
+                log.info("Job [{}] - contract [{}] ", jobUuid, contract.getContractNumber());
 
-                var jobOutputs = processContract(outputDir, contract);
+                var jobOutputs = processContract(outputDir, contract, jobUuid);
 
                 jobOutputs.forEach(jobOutput -> job.addJobOutput(jobOutput));
                 jobOutputRepository.saveAll(jobOutputs);
             }
 
             completeJob(job);
 
+        } catch (JobCancelledException e) {
+            job.setStatus(CANCELLED);
+            job.setStatusMessage(e.getMessage());
+            job.setExpiresAt(OffsetDateTime.now().plusDays(1));
+            jobRepository.save(job);
```

Comment:
We don't really need these. API has already updated the job record.

Reply:
removed